ha/k8s_setup/ha_setup.py

```diff
@@ -0,0 +1,342 @@
+        except Exception as c_err:
+            sys.stderr.write(f'HA config command failed: {c_err}.\n')
+
+class InitCmd(Cmd):
+    """
+    Init Setup Cmd
+    """
+    name = "init"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process init command.
+        """
+        sys.stdout.write('HA initialization is done.\n')
+
+class TestCmd(Cmd):
+    """
+    Test Cmd
+    """
+    name = "test"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process test command.
+        """
+        sys.stdout.write('Yet to be implemented...\n')
+
+class ResetCmd(Cmd):
+    """
+    Reset Cmd
+    """
+    name = "reset"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process reset command.
+        """
+        sys.stdout.write('HA reset is yet to be implemented...\n')
+
+class CleanupCmd(Cmd):
+    """
+    Cleanup Cmd
+    """
+    name = "cleanup"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process cleanup command.
+        """
+        Log.info("Processing cleanup command")
+        try:
+            # Delete the config file
+            self.remove_config_files()
+            if self._confstore is None:
+                self._confstore = ConfigManager.get_confstore()
+                self._confstore.delete(recurse=True)
+            Log.info("consul keys are deleted")
+            Log.info("cleanup command is successful")
+            sys.stdout.write("cleanup command is successful.\n")
+        except Exception as e:
+            Log.error(f"cleanup command failed. Error: {e}")
+            sys.stderr.write(f"cleanup command failed. {traceback.format_exc()}, Error: {e}\n")
+            raise HaCleanupException("cleanup failed")
+
+
+    def remove_config_files(self):
+        """
+        Remove file created by ha.
+        """
+        files = [const.CONFIG_DIR]
+
+        for file in files:
+            CleanupCmd.remove_file(file)
+        Log.info("All the config files are removed")
+
+def main(argv: list):
+    try:
+        if len(sys.argv) == 1:
+            Cmd.usage("ha_setup")
+            sys.exit(1)
+        if sys.argv[1] == "cleanup":
+            if not os.path.exists(const.HA_CONFIG_FILE):
+                a_str = f'Cleanup can not be proceed as \
+                           HA config file: {const.HA_CONFIG_FILE} \
+                           is missing. Either cleanup is already done or there \
+                           is some other problem'
+                sys.stdout.write(a_str)
+                return 0
+            ConfigManager.init("ha_setup")
```

Comment:
Why is init done in cleanup ?

Reply:
In init, we perform Log.init means logging initialization. So that's why its needed.